src/commands/Games/higherLower.ts

```diff
@@ -0,0 +1,267 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { SkyraCommand } from '../../lib/structures/SkyraCommand';
+import { Events } from '../../lib/types/Enums';
+import { UserSettings } from '../../lib/types/settings/UserSettings';
+import { Time } from '../../lib/util/constants';
+import { LLRCData, LongLivingReactionCollector } from '../../lib/util/LongLivingReactionCollector';
+import { getColor, resolveEmoji } from '../../lib/util/util';
+
+const enum HigherLowerReactions {
+	Higher = 'a:sarrow_up:658450971655012363',
+	Lower = 'a:sarrow_down:658452292558913536',
+	Cancel = ':redCross:637706251257511973',
+	Ok = ':greenTick:637706251253317669',
+	Cashout = '%F0%9F%92%B0' // 💰
+}
+
+const enum EndingAction {
+	Play,
+	Stop,
+	Timeout
+}
+
+export default class extends SkyraCommand {
+
+	private readonly kFirstReactionArray = [HigherLowerReactions.Higher, HigherLowerReactions.Lower, HigherLowerReactions.Cancel] as const;
+	private readonly kReactionArray = [HigherLowerReactions.Higher, HigherLowerReactions.Lower, HigherLowerReactions.Cashout] as const;
+	private readonly kWinReactionArray = [HigherLowerReactions.Ok, HigherLowerReactions.Cancel] as const;
+	private readonly kTimer = Time.Minute * 3;
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['hilo', 'higherlower', 'hl'],
+			bucket: 2,
+			cooldown: 7,
+			description: language => language.tget('COMMAND_HIGHERLOWER_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_HIGHERLOWER_EXTENDED'),
+			requiredPermissions: ['ADD_REACTIONS', 'MANAGE_MESSAGES', 'EMBED_LINKS', 'USE_EXTERNAL_EMOJIS'],
+			runIn: ['text'],
+			usage: '<50|100|200|500|1000|2000|5000|10000>'
+		});
+	}
+
+	public async run(message: KlasaMessage, [text]: [string]) {
+		await message.author.settings.sync();
+
+		const wager = Number(text);
+		const balance = message.author.settings.get(UserSettings.Money);
+		if (balance < wager) throw message.language.tget('GAMES_NOT_ENOUGH_MONEY', balance);
+
+		const response = await message.sendLocale('COMMAND_HIGHERLOWER_LOADING');
+		const game: HigherLowerGameData = {
+			/** The game's reaction collector */
+			llrc: new LongLivingReactionCollector(this.client, reaction => {
+				if (game.callback === null) return;
+
+				const emoji = this.resolveCollectedEmoji(message, game, reaction);
+				if (emoji === null) return;
+
+				game.callback(emoji);
+				game.callback = null;
+			}, async () => {
+				if (game.callback !== null) {
+					game.callback(null);
+					game.callback = null;
+				}
+				try {
+					await this.end(game, message, game.turn > 1);
+				} catch (error) {
+					this.client.emit(Events.Wtf, error);
+				}
+			}),
+			response,
+			running: true,
+			turn: 1,
+			number: this.random(0),
+			wager,
+			emojis: this.kFirstReactionArray,
+			callback: null
+		};
+
+		while (game.running) {
+			await game.response.reactions.removeAll();
```

Comment:
Is this really efficient? You're removing all reactions only to re-react... That's 1 DELETE, and then `n` PUTs for each emoji... versus at most one DELETE for the user reaction on an emoji...